src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,142 @@
+        {
+            'id': 'github_url',
+            'label': 'GitHub URL',
+            'type': 'string',
+            'help_text': gettext_noop(
+                'Specify the GitHub URL here in the case of an Enterprise Github '
+                'Example: https://example.github.com',
+            ),
+            'default': 'https://github.com',
+        },
+        {
+            'id': 'app_id',
+            'label': 'Github App id created by the GitHub Admin',
+            'type': 'string',
+            'help_text': gettext_noop(
+                'The GitHub App Id created by the GitHub Admin '
+                'Example App ID: 1121547 '
+                'from https://github.com/settings/apps/arestlel-github-app-ansible'
+                'required for creating a bearer token for authentication',
+            ),
+            'default': '1121547',
+        },
+        {
+            'id': 'install_id',
+            'label': 'Installation id',
+            'type': 'string',
+            'help_text': gettext_noop(
+                'The  Installation ID from the GitHub App installation generated by the GitHub Admin '
+                'Example: 59980338 '
+                'extracted from the installation link: https://github.com/settings/installations/59980338 '
+                'required for creating a limited github app token',
+            ),
+            'default': '59980338',
+        },
+        {
+            'id': 'ssh_key_data',
+            'label': gettext_noop('RSA Private Key'),
+            'type': 'string',
+            'format': 'ssh_private_key',
+            'secret': True,
+            'multiline': True,
+            'help_text': gettext_noop(
+                'Paste the contents of the PEM file that the GitHub Admin '
+                'provided to you with the app and installation ids.',
+            ),
+        },
+    ],
+    'required': [
+        'app_id',
+        'install_id',
+        'ssh_key_data',
+    ],
+}
+
+github_app_token = ManagedCredentialType(
+    namespace='github_app_token',
+    # scm source control management so it can be used for github
+    # authentication.
+    kind='scm',
+    name=gettext_noop('GitHub App Token'),
+    managed=True,
+    inputs=github_app_inputs,
+    custom_injectors=github_app_injector,
+)
+
+
+def github_app_backend(**kwargs):
+
+    github_url = kwargs.get('github_url')
+    app_id = kwargs.get('app_id')
+    installation_id = kwargs.get('install_id')
+    private_key = kwargs.get('ssh_key_data')
+    jwt_expiry = kwargs.get('jwt_expiry', 600)
+
+    missing_parameters = []
+    if not github_url:
+        missing_parameters.append('GitHub URL')
+    if not installation_id:
+        missing_parameters.append('Installation ID')
+    if not app_id:
+        missing_parameters.append('Application ID')
+    if not private_key:
+        missing_parameters.append('Private Key')
+    if missing_parameters:
+        raise Exception(f'Missing Parameter: {missing_parameters}')
```

Comment:
Is there an option to raise a specific exception that is not a catch-all and corresponds to the semantics of the situation in this code branch? `TypeError`? `ValueError`?